todo-app/src/App.js

```diff
@@ -1,10 +1,68 @@
 import React, { Component } from 'react';
+import "./style.css"
+import Header from "./components/Header.js"
+import TodoList from "./components/TodoList"
+import myTodoList from "./components/myTodoList.json"
+import moment from "moment"
+import AddingContainer from "./components/AddingContainer"
+
 
 class App extends Component {
+  state = {
+    tasks: myTodoList
+  }
+  componentWillUpdate(nextProps, nextState) {
+    localStorage.setItem("todo list", JSON.stringify(nextState.tasks))
+  }
+  componentDidMount() {
+    if (localStorage.getItem("todo list")) {
+      this.setState({ tasks: JSON.parse(localStorage.getItem("todo list")) })
+    }
+  }
+  nextId = () => {
+    if (this.state.tasks.length === 0) {
+      return 1;
+    }
+    let uniqueId = this.state.tasks[0].id;
+    return uniqueId += 1;
+  }
+  addingTask = (description, deadLine) => {
+    this.setState((previousState) => ({
+      tasks: [{ id: this.nextId(), description: description, deadLine: deadLine, done: false, creatingDate: moment().format("llll") }, ...previousState.tasks]
+    })
+    )
+  }
+  handleCheckingBox = (taskId) => {
```

Comment:
First of all, you are mutating object properties directly instead - you are unintentionally making mutations to your state. Second, `map` exists exactly for this reason, instead of declaring an array and then doing a `forEach`. A better way to write this code would be like:

    handleCheckingBox = (taskId) => {
      const newState = this.state.tasks.map(task => {

        // Returning a _new_ object so that we don't mutate objects inside our state
        return {
          ...task,
          done: taskId === task.id ? !task.done : task.done,
        }
      })

    }